src/array.cr

```diff
@@ -1201,6 +1201,13 @@ class Array(T)
     pop { raise IndexError.new }
   end
 
+  # Removes the last value from `self`.
+  # If the array is of size 0, the given block is called.
+  #
+  # ```
+  # a = [1]
+  # a.pop { "Testing" } #=> 1
+  # a.pop { "Testing" } #=> "Testing"
   def pop
```

Comment:
Yes, it's public. Maybe you'd like to pop a value or use another default value instead of `nil`